simulator/src/message.rs

```diff
@@ -38,7 +39,6 @@ impl<'a> TraceMessage {
         .attributes
     }
 
-    #[tracing::instrument]
```

Comment:
Why have these been removed? Assuming they create spans (which was my understanding) it seems like a low effort way to get more fine grained information.

As a side note, assuming my assumptions are correct, I would prefer if this was used as the main way to instrument the code for tracing going forward given that it promotes appropriately sized, well organised functions.